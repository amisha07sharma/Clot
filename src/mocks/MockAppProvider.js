import React, { useReducer } from "react";
import PropTypes from "prop-types";
import { AppContext } from "../context/AppContext";
import appReducer from "../reducers/appReducer";

const initialState = {
  email: "mocked_email@example.com",
  firstName: "John",
  lastName: "Doe",
};

const MockAppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(appReducer, initialState);

  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

MockAppProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MockAppProvider;
