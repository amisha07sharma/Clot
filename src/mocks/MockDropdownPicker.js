import React from "react";
import PropTypes from "prop-types";
import { Text, TouchableOpacity } from "react-native";

const MockDropDownPicker = ({
  open,
  value,
  items,
  setOpen,
  setValue,
  setItems,
  placeholder,
  dropDownContainerStyle,
  textStyle,
}) => {
  return (
    <TouchableOpacity
      style={dropDownContainerStyle}
      onPress={() => setOpen(!open)}
      testID="Dropdown"
    >
      <Text>{value ? value.label : placeholder}</Text>
    </TouchableOpacity>
  );
};

MockDropDownPicker.propTypes = {
  open: PropTypes.bool,
  value: PropTypes.object,
  items: PropTypes.array,
  setOpen: PropTypes.func,
  setValue: PropTypes.func,
  setItems: PropTypes.func,
  placeholder: PropTypes.string,
  dropDownContainerStyle: PropTypes.object,
  textStyle: PropTypes.object,
};

export default MockDropDownPicker;
