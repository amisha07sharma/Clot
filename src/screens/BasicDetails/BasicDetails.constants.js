export const ageRange = [
  { label: "0-5", value: 1 },
  { label: "5-18", value: 2 },
  { label: "18-25", value: 3 },
  { label: "25-40", value: 4 },
  { label: "above 40", value: 5 },
];
