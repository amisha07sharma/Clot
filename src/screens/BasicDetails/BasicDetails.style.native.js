import styled from "styled-components/native";
import { Text } from "react-native";
import DropDownPicker from "react-native-dropdown-picker";
import { PURPLE, LIGHT_GREY, WHITE } from "../../constants/colors";

export const CustomWrapper = styled.View`
  flex: 1;
  justify-content: space-between;
`;

export const TopSection = styled.View`
  flex: 0.97;
  background-color: ${WHITE};
  padding: 20px;
`;

export const BottomSection = styled.View`
  background-color: ${LIGHT_GREY};
  padding-horizontal: 20px;
  padding-bottom: 20px;
`;

export const ContentWrapper = styled.View`
  flex-direction: row;
  justify-content: space-between;
  margin-bottom: 50px;
`;

export const CustomButtonWrapper = styled.View`
  background-color: ${(props) => (props.isActive ? PURPLE : LIGHT_GREY)};
  height: 45px;
  width: 160px;
  border-radius: 100px;
  justify-content: center;
`;

export const TextContainer = styled(Text)`
  font-size: 16px;
  font-weight: 450;
  line-height: 20.24px;
  text-align: left;
  margin-bottom: 20px;
`;

export const DropDownPickerContainer = styled(DropDownPicker)`
  border-radius: 100px;
  background-color: ${LIGHT_GREY};
  border-width: 0px;
  padding-horizontal: 20px;
`;
