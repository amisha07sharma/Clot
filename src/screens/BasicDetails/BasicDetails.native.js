import React, { useState } from "react";
import PropTypes from "prop-types";
import CLOT from "../../constants/messages.json";
import CustomButton from "../../components/CustomButton";
import { HeadingText } from "../../styles/common.style";
import { ageRange } from "./BasicDetails.constants";
import RadioButton from "./RadioButton";
import {
  CustomWrapper,
  TopSection,
  BottomSection,
  ContentWrapper,
  TextContainer,
  DropDownPickerContainer,
} from "./BasicDetails.style";

const BasicDetails = ({ onContinue }) => {
  const [isActive, setIsActive] = useState(true);
  const [open, setOpen] = useState(false);
  const [value, setValue] = useState(null);
  const [items, setItems] = useState(ageRange);

  return (
    <CustomWrapper>
      <TopSection>
        <HeadingText>{CLOT.BASIC_DETAILS_SCREEN.ABOUT_YOURSELF}</HeadingText>
        <TextContainer>{CLOT.BASIC_DETAILS_SCREEN.SHOP_FOR}</TextContainer>
        <ContentWrapper>
          <RadioButton
            testID="Men"
            title={CLOT.BASIC_DETAILS_SCREEN.MEN}
            onPress={() => {
              setIsActive(true);
            }}
            isActive={isActive}
          />
          <RadioButton
            testID="Women"
            title={CLOT.BASIC_DETAILS_SCREEN.WOMEN}
            onPress={() => {
              setIsActive(false);
            }}
            isActive={!isActive}
          />
        </ContentWrapper>
        <TextContainer>{CLOT.BASIC_DETAILS_SCREEN.HOW_OLD}</TextContainer>
        <DropDownPickerContainer
          testID="DropDown"
          open={open}
          value={value}
          items={items}
          setOpen={setOpen}
          setValue={setValue}
          setItems={setItems}
          dropDownContainerStyle={{
            borderWidth: 0.2,
          }}
          textStyle={{
            fontSize: 16,
            fontWeight: 450,
            lineHeight: 20.24,
            textAlign: "left",
          }}
        />
      </TopSection>
      <BottomSection>
        <CustomButton
          testID="ContinueButton"
          title={CLOT.BASIC_DETAILS_SCREEN.FINISH}
          onPress={onContinue}
          disabled={!value}
        />
      </BottomSection>
    </CustomWrapper>
  );
};

BasicDetails.propTypes = {
  onContinue: PropTypes.func.isRequired,
};

export default BasicDetails;
