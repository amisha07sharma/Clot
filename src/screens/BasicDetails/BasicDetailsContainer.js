import React from "react";
import PropTypes from "prop-types";
import clot from "../../api/clot";
import BasicDetails from "./BasicDetails";

const BasicDetailsContainer = ({ navigation }) => {
  const onContinue = () => {
    clot
      .post("/auth/login", {
        username: "mor_2314",
        password: "83r5^_",
      })
      .then((response) => {
        console.log("api call", response.data);
        navigation.navigate("Tabs");
      })
      .catch((err) => {
        console.error(err);
      });
  };
  return <BasicDetails onContinue={onContinue} testID="BasicDetails" />;
};

BasicDetailsContainer.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default BasicDetailsContainer;
