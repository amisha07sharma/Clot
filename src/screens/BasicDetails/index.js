import BasicDetailsContainer from "./BasicDetailsContainer";

export default BasicDetailsContainer;
