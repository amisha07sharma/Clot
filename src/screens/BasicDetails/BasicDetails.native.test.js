import React from "react";
import { View as MockView } from "react-native";
import { render, screen, fireEvent } from "@testing-library/react-native";
import BasicDetails from "./BasicDetails";

jest.mock("react-native-dropdown-picker", () =>
  require("../../mocks/MockDropdownPicker")
);

jest.mock("./RadioButton", () => (props) => (
  <MockView {...props} testID={props.testID} />
));

jest.mock("../../components/CustomButton", () => (props) => (
  <MockView {...props} />
));

const mockOnContinue = jest.fn();
const renderComponent = (props) => {
  render(<BasicDetails onContinue={mockOnContinue} {...props} />);
};
describe("BasicDetails Component", () => {
  it("should render BasicDetails component correctly", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should toggle between men and women", () => {
    renderComponent();

    expect(screen.getByTestId("Men").props.isActive).toBe(true);

    fireEvent.press(screen.getByTestId("Women"));

    expect(screen.getByTestId("Women").props.isActive).toBe(true);
    expect(screen.getByTestId("Men").props.isActive).toBe(false);
  });

  it("should interact with mocked DropDownPicker", () => {
    renderComponent();

    fireEvent.press(screen.getByTestId("Dropdown"));
  });

  it("should call onContinue prop when button is pressed", () => {
    renderComponent();

    fireEvent.press(screen.getByTestId("ContinueButton"));

    expect(mockOnContinue).toHaveBeenCalledTimes(1);
  });
});
