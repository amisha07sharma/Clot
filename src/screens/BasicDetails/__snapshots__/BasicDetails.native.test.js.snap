// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicDetails Component should render BasicDetails component correctly 1`] = `undefined`;
